Sources/AsyncHTTPClient/HTTPHandler.swift

```diff
@@ -765,12 +765,32 @@ extension TaskHandler: ChannelDuplexHandler {
             return context.eventLoop.makeSucceededFuture(())
         }
 
-        return body.stream(HTTPClient.Body.StreamWriter { part in
-            context.eventLoop.assertInEventLoop()
-            return context.writeAndFlush(self.wrapOutboundOut(.body(part))).map {
-                self.callOutToDelegateFireAndForget(value: part, self.delegate.didSendRequestPart)
+        func doIt() -> EventLoopFuture<Void> {
+            return body.stream(HTTPClient.Body.StreamWriter { part in
+                let promise = self.task.eventLoop.makePromise(of: Void.self)
```

Comment:
this whole block should just be

```swift
return channel.writeAndFlush(HTTPClientRequestPart.body(part)).map {
    self.callOutToDelegateFireAndForget(value: part, self.delegate.didSendRequetsPart)
}
```

Reply:
we only have access to `context` here, so in order to get to `channel` we have to call `context.channel` that is precondition on an being on channel EL, am I missing something?

Reply:
actually, even if I get the channel, I cannot write `.part` to it since first handler in the channel expects `Request`